### ISSUE TITLE
False Positive UnusedPrivateClass

### ISSUE THREAD

**Issue by scottkennedy** [arturbosch/detekt#1474]
## Expected Behavior
No UnusedPrivateClass

## Observed Behavior
UnusedPrivateClass - `InternalClass`

## Steps to Reproduce
```kt
private class DetektFalsePositiveTest {
    fun getSomeObject(): ((String) -> Any) = ::InternalClass

    private class InternalClass(val param: String)
}
```

`InternalClass` is clearly used in `getSomeObject()`, but it's being reported unused.

## Your Environment
* Version of detekt used: RC13
* Version of Gradle used (if applicable): 5.2.1
* Operating System and version: Various

**Comment by schalkms** [comment]
Thanks for reporting this.
It seems like this rule doesn't handle reflection references at the moment.

**Comment by arturbosch** [comment]
Unfortunately this one is tricky without proper type resolution :(.
Currently we are inspecting the reiceiver of a double colon expressions and treat the right side as a function call. However this example tells it is a constructor call, so this false positive is triggered.
Without type resolution we can track the declaration for each scope.

**Comment by arturbosch** [comment]
Again pushed to master while being in another branch ... idea has some serious bugs here lately ...

**Comment by vanniktech** [comment]
@arturbosch let's enforce CI checks on master branch that way even if you push the commit won't be accepted. It's considered good practice anyways.

**Comment by arturbosch** [comment]
That was the missing piece - CI checks, thanks!
